src/back/controllers/actions/user.js

```diff
@@ -21,7 +21,29 @@ function watchChallenge(userPromise, challengeId) {
     return userPromise.then(user => {
         user.set(
             'watchList',
-            user.get('watchList').reduce((list, id) => [...list, id], [challengeId])
+            user
+                .get('watchList')
+                .filter(taskId => taskId.toString() !== challengeId)
+                .reduce((list, id) => [...list, id], [challengeId])
+        );
+        return user.save();
+    });
+}
+
+function acceptChallenge(userPromise, challengeId) {
+    const STATUS_ACCEPTED = 'accepted';
+    const newlyAccepted = {
+        status: STATUS_ACCEPTED,
+        date: new Date(),
+        challenge: challengeId
+    };
+    return userPromise.then(user => {
+        user.set(
+            'challenges',
+            user
+                .get('challenges')
+                .filter(task => task.status !== STATUS_ACCEPTED && challengeId !== task.challenge)
+                .reduce((list, task) => [...list, task], newlyAccepted)
```

Comment:
bug arrised, because of this line cause reduce returns newlyAccepted extended with values of last filtered challenge